### DISCUSSION TITLE
Unset environment variables when running app on Production mode - using config from object method

### DISCUSSION THREAD

**Discussion by ArsHinMQ** [miguelgrinberg/Flask-SocketIO#1710] [Q&A]
So, I created a Flask application and I used Flask-SocketIO is a part of it, it worked fine until I removed the ```debug=True``` under ```if __name__ == "__main__"```
app.py:
```
from flask import Flask
from dotenv import load_dotenv, find_dotenv
from flask_socketio import SocketIO

from config import Config

load_dotenv(find_dotenv())

def create_app():
    app = Flask(__name__)
    app.config.from_object(Config)

    return app

app = create_app()
io = SocketIO(app, engineio_logger=True)

# Registered Blueprints and imported socket's events

if __name__ == "__main__":
    io.run(app)
```
Now it gave me 500 error:
```RuntimeError: A secret key is required to use CSRF.```
Am I missing something?
What is the best way to set Flask config variables when using Flask-SocketIO?

**Comment by miguelgrinberg** [comment]
This has nothing to do with Flask-SocketIO, that error is coming from another extension, possibly Flask-WTF. Have you considered doing what the error says? It's simple enough to add a secret key.

**Reply by ArsHinMQ** [reply]
I have a SECRET_KEY, in my .env file, and I get it in config.py file
config,py:
```
from os import getenv

class Config:
    # Flask requirements
    SECRET_KEY = getenv("SECRET_KEY")
```
and it works when I set ```debug=True```
so I just thought it can be Flask-SocketIO.

**Reply by miguelgrinberg** [reply]
You have to debug this, because the error indicates that Flask does not have the secret key configured.
